tests/Unit/PointwiseFunctions/GeneralRelativity/Test_ComputeGhQuantities.cpp

```diff
@@ -186,22 +191,180 @@ void test_lapse_deriv_functions_analytic(
+  const auto& dt_shift_expected =
+    get<Tags::dt<gr::Tags::Shift<SpatialDim>>>(vars);
+  const auto& spatial_metric = get<gr::Tags::SpatialMetric<SpatialDim>>(vars);
+  const auto& dt_spatial_metric =
+      get<Tags::dt<gr::Tags::SpatialMetric<SpatialDim>>>(vars);
+  const auto& d_spatial_metric =
+      get<typename Solution::template DerivSpatialMetric<DataVector>>(vars);
+
+  // Get ingredients
+  const auto inverse_spatial_metric =
+      determinant_and_inverse(spatial_metric).second;
+  const auto inverse_spacetime_metric =
+      gr::inverse_spacetime_metric(lapse, shift, inverse_spatial_metric);
+  const auto phi = GeneralizedHarmonic::phi(lapse, d_lapse, shift,
+    d_shift_expected, spatial_metric, d_spatial_metric);
+  const auto pi = GeneralizedHarmonic::pi(lapse, dt_lapse, shift,
+    dt_shift_expected, spatial_metric, dt_spatial_metric, phi);
+  const auto normal_vector = gr::spacetime_normal_vector(lapse, shift);
+  const auto lower_shift = raise_or_lower_index(shift, spatial_metric);
+
+  // Compute dt_lower_shift
+  auto dt_lower_shift_expected =
+    make_with_value<tnsr::i<DataVector, SpatialDim, Frame::Inertial>>(x, 0.);
+  for (size_t i = 0; i < SpatialDim; ++i) {
+    for (size_t j = 0; j < SpatialDim; ++j) {
+      dt_lower_shift_expected.get(i) += spatial_metric.get(i, j) *
+        dt_shift_expected.get(j) + shift.get(j) * dt_spatial_metric.get(i, j);
+    }
+  }
+  // Compute d4_norm_shift
+  auto d4_norm_of_shift_expected =
+    make_with_value<tnsr::a<DataVector, SpatialDim, Frame::Inertial>>(x, 0.);
+  // compute intermediate quantities
+  for (size_t i = 0; i < SpatialDim; ++i) {
+    // first term for component 0
+    get<0>(d4_norm_of_shift_expected) += lower_shift.get(i) *
+      dt_shift_expected.get(i) + shift.get(i) * dt_lower_shift_expected.get(i);
+    // second term for components 1,2,3
+    for (size_t j = 0; j < SpatialDim; ++j) {
+      d4_norm_of_shift_expected.get(1+j) += lower_shift.get(i)
+        * d_shift_expected.get(j, i) + shift.get(i) * phi.get(j, 0, i+1);
+    }
+  }
+
+  // Check that locally computed derivs match returned ones
+  const auto d0_shift = GeneralizedHarmonic::time_deriv_of_shift<
+    SpatialDim, Frame::Inertial, DataVector>(lapse, shift,
+      inverse_spatial_metric, normal_vector, phi, pi);
+
+  const auto d_shift = GeneralizedHarmonic::spatial_deriv_of_shift<
+    SpatialDim, Frame::Inertial, DataVector>(lapse, inverse_spacetime_metric,
+      normal_vector, phi);
+
+  const auto d0_lower_shift =
+    GeneralizedHarmonic::time_deriv_of_lower_shift<SpatialDim, Frame::Inertial,
+      DataVector>(lapse, shift, spatial_metric, normal_vector, phi, pi);
+
+  const auto d4_norm_shift =
+    GeneralizedHarmonic::spacetime_deriv_of_norm_of_shift<SpatialDim,
+      Frame::Inertial, DataVector>(lapse, shift, spatial_metric,
+        inverse_spatial_metric, inverse_spacetime_metric, normal_vector, phi,
+          pi);
+
+  CHECK_ITERABLE_APPROX(dt_shift_expected, d0_shift);
```

Comment:
same as above with `dt` and `d0` (and on line 366)

Reply:
fixed